### PR TITLE
Add support for metadata.

### LINKED ISSUE #16: Add support for Box metadata

### PR DESCRIPTION
**Pull request by Jeff-Meadows**
Fixes #16.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @Jeff-Meadows has signed the CLA. Thanks for the pull request!

File: README.rst [hunk 1]
```diff
@@ -169,6 +169,23 @@ Events
...
+.. code-block:: python
+
+    # Get metadata
+    client.file(file_id='SOME_FILE_ID').metadata().get()
+
+    # Create metadata
+    client.file(file_id='SOME_FILE_ID').metadata().create({'key', 'value')
+
+    # Update metadata
+    from boxsdk.object.metadata import MetadataUpdate
```
Comment (aptxkid) [inline README.rst]: Should we start putting classes for external access to the __init__ .py of the boxsdk package? Then users can do `from boxsdk import MetadataUpdate`.

File: boxsdk/object/file.py [hunk 2]
```diff
@@ -1,10 +1,9 @@
 # coding: utf-8
 
 from __future__ import unicode_literals
-
 from boxsdk.config import API
-
-from .item import Item
+from boxsdk.object.item import Item
```
Comment (aptxkid) [inline boxsdk/object/file.py]: I think we should at least be consistent about if we are going to import with full path or relative path.

File: boxsdk/object/metadata.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        :type value:
+            `unicode`
+        :param old_value:
+            If specified, only update the key/value pair if the current value is equal to oldValue.
+        :type old_value:
+            `unicode`
+        """
+        if old_value is not None:
+            self._ops.append({'op': 'test', 'path': path, 'value': old_value})
+        self._ops.append({'op': 'replace', 'path': path, 'value': value})
```
Comment (aptxkid) [inline boxsdk/object/metadata.py]: I guess you can merge all these operation methods into one.

**Comment by aptxkid** [comment]
Added some minor comments. Otherwise, LGTM :+1:

File: boxsdk/object/file.py (hunk 2 above)
Comment (Jeff-Meadows) [inline boxsdk/object/file.py]: Yep. I think we should use absolute, but we can update that in a different PR.

File: README.rst (hunk 1 above)
Comment (Jeff-Meadows) [inline README.rst]: Yeah, it shouldn't be so hard to do an update. Either from boxsdk import MetadataUpdate, or, perhaps better, add a method to Metadata that will give you a MetadataUpdate object.

**Pull request merged**